ballerina/listener.bal

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2025 WSO2 LLC (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+
+# Represents the options for configuring an MCP server.
+public type ServerOptions record {|
+    # Capabilities to advertise as being supported by this server.
+    ServerCapabilities capabilities?;
+    # Optional instructions describing how to use the server and its features.
+    string instructions?;
+    # Whether to enforce strict capabilities compliance.
+    boolean enforceStrictCapabilities?;
+|};
+
+# Configuration options for initializing an MCP listener.
+public type ListenerConfiguration record {|
+    *http:ListenerConfiguration;
+    *ServerConfiguration;
+|};
+
+type ServerConfiguration record {|
+    Implementation serverInfo;
+    ServerOptions options?;
+|};
+
+# A server listener for handling MCP service requests.
+public class Listener {
```

Comment:
shall we add a test case by implementing an mcp service and consuming it using client?